src/python-common/ceph/deployment/service_spec.py

```diff
@@ -1095,3 +1113,106 @@ def get_port(self) -> int:
             return {'prometheus': 9095,
                     'node-exporter': 9100,
                     'grafana': 3000}[self.service_type]
+
+
+class SNMPGatewaySpec(ServiceSpec):
+
+    def __init__(self,
+                 service_type: str = 'snmp-gateway',
+                 snmp_version: Optional[str] = None,
+                 snmp_destination: Optional[str] = None,
+                 credentials: Optional[Dict[str, str]] = None,
+                 engine_id: Optional[str] = None,
+                 auth_protocol: Optional[str] = None,
+                 privacy_protocol: Optional[str] = None,
+                 placement: Optional[PlacementSpec] = None,
+                 unmanaged: bool = False,
+                 preview_only: bool = False,
+                 port: Optional[int] = 9464,
+                ):
+        assert service_type == 'snmp-gateway'
+        assert snmp_version is not None
```

Comment:
can we make a proper default value here?

Reply:
default for snmp version? Yeah that makes sense - not sure what most entperprise users are using though - that's why I'm not defaulting in this PR. Perhaps something to change once we have users trying this out?